source/slang/slang-ir.cpp

```diff
@@ -6132,6 +6132,18 @@ namespace Slang
         return i;
     }
 
+    // IR emitter for a dedicated instruction to represent NonUniformResourceIndex qualifier.
+    IRInst* IRBuilder::emitNonUniformResourceIndex(IRType* type, IRInst* val)
```

Comment:
Can simply things by deriving type from inst instead of asking the caller to provide the type.

Reply:
done